src/Data/Unrestricted/Internal/Instances.hs

```diff
@@ -351,3 +361,74 @@ instance (Movable a, Prelude.Semigroup a) => Semigroup (MovableMonoid a) where
       combine (Ur x) (Ur y) = x Prelude.<> y
 
 instance (Movable a, Prelude.Monoid a) => Monoid (MovableMonoid a)
+
+instance Consumable (RepStream a) where
+  consume (RepStream givex dupsx consumesx sx) = consumesx sx
+instance Dupable (RepStream a) where
+  dupR (RepStream givex dupsx consumesx sx) = CollectFrom $ RepStream
+    (\sx' -> RepStream givex dupsx consumesx sx')
+    dupsx
+    consumesx
+    sx
+
+instance Consumable (Replicator a) where
+  consume = \case
+    Moved _ -> ()
+    CollectFrom stream -> consume stream
+
+instance Dupable (Replicator a) where
+  dupR = \case
+    Moved x -> Moved (Moved x)
+    CollectFrom stream -> CollectFrom Data.<$> dupR stream
+
+instance Data.Functor RepStream where
+  fmap f (RepStream givef dupsf consumesf sf) =
+    RepStream (f . givef) dupsf consumesf sf
+instance Data.Applicative RepStream where
```

Comment:
These instances should be in the module where `Functor` and `Applicative` are defined, I think.